#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9999872709942303 -0.0034700016550398734 0.0036629138709121487 1.3043651460926227
  position 0.18968417740182222 8.122375547720665 3.513966176160681
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Microbot {
  hidden position_0_0 -4500.053822481178
  hidden position_0_1 -4384.440128211954
  hidden linearVelocity_0 -0.12498187426482037 3.1027687993076893e-08 -0.0003995669331703265
  hidden angularVelocity_0 5.39102087711521e-06 -6.157026292496028e-05 -1.018496446033563e-08
  hidden translation_7 1.7982362802149154e-05 4.168239606435614e-05 0.002101587463776866
  hidden rotation_7 -0.12589310467918557 0.9877729628871964 0.09195379265307545 -2.0154576923963277e-08
  hidden linearVelocity_7 -0.12498361074418028 -1.2051350670605233e-07 -0.00040269604519813687
  hidden angularVelocity_7 5.391020862100297e-06 -6.157026300853926e-05 -1.0184965105456568e-08
  hidden translation_8 -1.1707529498306002e-05 4.2350956935638395e-05 0.0020981833170546915
  hidden rotation_8 0.10065794616695285 0.988601385293844 0.11196105960798794 1.609453195506102e-08
  hidden linearVelocity_8 -0.12498011827557522 1.8527905975545058e-07 -0.0004026844779702983
  hidden angularVelocity_8 5.391020866394214e-06 -6.157026300916831e-05 -1.0184969556624447e-08
  hidden rotation_10 1 0 0 1.2931425405941137
  hidden linearVelocity_10 -0.12498295912330461 1.0400172184073464e-07 -0.0004002268674017567
  hidden angularVelocity_10 -0.03203828433264471 -0.00012712756728641065 9.999948649884667
  hidden rotation_11 -1 0 0 1.2232161993970727
  hidden linearVelocity_11 -0.12498078606380232 -4.0777252080404215e-08 -0.00040017549811230494
  hidden angularVelocity_11 -0.03203809669239416 -0.00012721308035508407 9.999948636246554
  translation -0.9576695669471307 0.021503260504657784 1.3704995851529282
  rotation -0.001314742945657849 0.9999982718754561 -0.0013144189164992182 -1.5740010557309552
  controller "wall_follower"
  extensionSlot [
    DistanceSensor {
      translation 0.05 0 0.05
      name "left"
    }
    DistanceSensor {
      translation -0.05 -1.22461e-18 0.05
      rotation 0 0 1 3.141592653589793
      name "right"
    }
    DistanceSensor {
      translation 0 -7.34766e-18 0.06
      rotation 0 1 0 -1.5707950010791611
      name "middle"
    }
  ]
}
RectangleArena {
  floorSize 5 5
  floorAppearance Parquetry {
    type "light strip"
  }
}
Solid {
  translation -1.2576345376694211 0.039708387633600496 1.561774593116869
  rotation -0.9999999796163702 0.0001941420150753747 5.5462934606849e-05 -1.643513213806917
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2.5 0.1 0.1
      }
    }
  ]
  name "solid1"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity 0.00045798991222052916 6.192069085758482e-05 2.3604739903289634e-05
  angularVelocity -0.00411939812137144 4.2820553259870785e-06 1.5142713319757796e-05
}
Solid {
  translation -1.5 0.042151628561723435 0.13
  rotation -0.8082939444548717 0.5886443204051871 0.012600135408025433 2.0299261788373305e-16
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.1 0.1
      }
    }
  ]
  name "solid2"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity 3.652391627524599e-18 3.0593511894425266e-06 -2.022322162603887e-17
  angularVelocity -4.629198785436405e-16 1.099860811809646e-17 4.8498126995995304e-17
}
Solid {
  translation -1.2689901955086609 0.04642631424182119 1.0156382158486659
  rotation 0.0016501168315630826 0.99999826452154 0.0008649094463222191 -1.559142212825335
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1 0.1 0.1
      }
    }
  ]
  name "solid1(2)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity -4.113939764498297e-06 -4.638648675081639e-05 -0.00022380330684620725
  angularVelocity 0.00010264279674144734 2.92729389103137e-05 0.0003760448314306692
}
Solid {
  translation -1.0060814816321384 0.03845637714944146 -1.3976139976183541
  rotation 1.557611460327383e-05 0.999889283458593 -0.014880207728584142 -3.1258341866683064
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 3 0.1 0.1
      }
    }
  ]
  name "solid1(3)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity 0.0003599098072435985 -3.487773622424962e-05 1.353189748030748e-05
  angularVelocity 0.0016510430195981003 1.5440334135941888e-05 1.0138631881714833e-05
}
Solid {
  translation 0.43882666159051387 0.04015256581784438 -0.12038770806276539
  rotation -0.5772119308618395 0.5770742669628661 -0.5777643786873831 -2.0955851136727612
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2.5 0.1 0.1
      }
    }
  ]
  name "solid1(4)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity 6.254732419470913e-06 -5.938462666381384e-07 4.4420123372462974e-05
  angularVelocity 2.4684844467873744e-07 -2.6863033710480417e-06 -3.7957991630178086e-05
}
Solid {
  translation 1.0806405604489273 0.0459082168988771 -2.000363840455008
  rotation -0.5766794080892059 0.5778599812390506 -0.577510781170609 -2.0940376284173716
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1 0.1 0.1
      }
    }
  ]
  name "solid1(1)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity -1.1980776825271097e-05 3.116838433884367e-05 -0.0001406591442267867
  angularVelocity 5.490973161682162e-05 -1.910781898125957e-05 3.317354846423172e-05
}
Solid {
  translation 1.7558130829239535 0.043459011861021725 -1.46443807457663
  rotation -0.9999323990744383 -0.007940168178306162 0.008494175094646492 -1.605440775565526
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1.5 0.1 0.1
      }
    }
  ]
  name "solid1(5)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity -0.00034453258384501665 2.203306328010512e-05 0.00016138112107530933
  angularVelocity -2.9330909900774476e-05 0.00020268879898690327 7.090131206321986e-05
}
Solid {
  translation 1.7545047327568484 0.04384467181418924 -0.300567683612741
  rotation -0.991963384421536 0.09158689639563275 -0.08729538576343931 -1.5321011379507123
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1.5 0.1 0.1
      }
    }
  ]
  name "solid1(6)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity -5.215822703726209e-07 -3.389439376175394e-05 -0.00033442043199744063
  angularVelocity 0.0017911231237044936 0.00047548158846976606 0.0002273354438802852
}
Solid {
  translation 1.6035525219328186 0.038001079683169556 1.1719877281599147
  rotation -0.38492995447703154 0.6506826359249818 -0.65455407526959 -2.411545924173142
  children [
    DEF b1 Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 3 0.1 0.1
      }
    }
  ]
  name "solid1(7)"
  boundingObject USE b1
  physics Physics {
  }
  linearVelocity 4.24264691086717e-05 -3.650188288399173e-05 6.737635555374102e-05
  angularVelocity 0.0007917074907901472 -8.289597796164095e-05 -0.0016905913324090117
}
